public/app/core/utils/shortLinks.ts

```diff
@@ -18,17 +20,36 @@ function buildHostUrl() {
   return `${window.location.protocol}//${window.location.host}${config.appSubUrl}`;
 }
 
+export function buildShortUrl(k8sShortUrl: ShortURL) {
+  const key = k8sShortUrl.metadata.name;
+  const orgId = k8sShortUrl.metadata.namespace;
+  return `${window.location.protocol}//${window.location.host}${config.appSubUrl}/goto/${key}?orgId=${orgId}`;
+}
+
 function getRelativeURLPath(url: string) {
   let path = url.replace(buildHostUrl(), '');
   return path.startsWith('/') ? path.substring(1, path.length) : path;
 }
 
 export const createShortLink = memoizeOne(async function (path: string) {
   try {
-    const shortLink = await getBackendSrv().post(`/api/short-urls`, {
-      path: getRelativeURLPath(path),
-    });
-    return shortLink.url;
+    if (config.featureToggles.useKubernetesShortURLsAPI) {
```

Comment:
This is the most important change in this PR (other than autogenerated code, new feature toggle and tests). When FF is enabled, we make a request to the new `shorturl.grafana.app` api to create a new short url.